llvm/lib/Target/X86/GISel/X86InstructionSelector.cpp

```diff
@@ -1789,6 +1793,47 @@ bool X86InstructionSelector::selectMulDivRem(MachineInstr &I,
   return true;
 }
 
+bool X86InstructionSelector::selectSelect(MachineInstr &I,
```

Comment:
Is there a reason this isn't using tablegen? As far as I can tell the DAG is not manually selecting this

Reply:
Yes. There are only patterns for `X86ISD::CMOV` so we need to copy-paste patterns for `ISD::SELECT`.
There is another way: we can legalize `G_SELECT` into `G_CMOV` but then we'll meet a problem that there is no register bank for `EFLAGS`. So we can't match instructions as we can't match with EFLAGS. It seems that most of the opcodes here are related to `EFLAGS`. This fact I've revealed only now. For sure it should be fixed so that EFLAGS matching works and we can reuse tblgen.

Reply:
Yes, I would expect you would aim to replicate what the DAG does and use similar looking intermediates

Reply:
@arsenm Yes, I completely agree with you. Would you insist that a register bank for EFLAGS should be added prior G_SELECT?